Day0/krischal111/assignment2/solution.c

```diff
@@ -0,0 +1,104 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+#define expressionfile "expressions.txt"
+#define resultfile "result.txt"
+
+int getnum(FILE * myfile)
+{
+    char numberstring[10] = "         ";
+    
+    // removing whitespace at the beginning
+    char c = ' ';
+    while(c == ' ')
+        c = fgetc(myfile);
+        
+    // extracting number as string into a variable
+    for(int i = 0; i<10; i++)
+    {
+        numberstring[i] = c;
+        
+        // break when numbers are completed
+        if(c < '0' || c > '9')
+        {
+            fseek(myfile, -1, 1); // Ek step paxadi aija vaneko
+            break;
+        }
+        c = fgetc(myfile);
+    }
+    
+    // return the number in our string
+    return atoi(numberstring);
+}
+
+char getoperator(FILE* myfile)
+{
+    char c;
+    do
+    {
+        c = fgetc(myfile);
+    } while (c == ' ') ;
+    
+    return c;
+}
+
+int main()
+{
+    FILE * myfile = fopen(expressionfile, "r");
+    if(myfile)
+    {
+	FILE * resfile = fopen(resultfile, "w");
+	fclose(resfile);
+	
+	while( 1 )
+	{  
+	    char c = fgetc(myfile);
+	    fseek(myfile, -1, 1);
+	    
+	    if(c == EOF)
+	        break;
+	
+	    float a = getnum(myfile);
+	    while(c != '\n' && c != EOF)
+	    {
+	        int b = getnum(myfile);
+	        
+	        switch(getoperator(myfile))
```

Comment:
This program doesn't work for all postfix expressions, such as "5 6 7 + *"

Reply:
Resolved (Had to learn, make and use custom stack, but works now + also shows calculation steps)